Scripts/generate-test-config.sh

```diff
@@ -1,14 +1,269 @@
 #!/bin/bash
```

Comment:
> I'm slightly concerned with the complexity of the script and wonder if we're trying too hard to make this automatic. It might be better to just tell people you need boto3 and Python and leave it at that, but, if this portably works and doesn't create problems, 👍!

Regarding `python3`, `aws` and `jq`: There's no automatic installation of these dependency utilities, just validation that they exist, which is something that we would have to do anyway or face weird errors during invocation. 

Regarding `boto3` and any other deps in the `requirements.txt`: I'm far from a Python expert, but the `venv` pattern here seems to be a pretty common way of encapsulating dependencies without polluting a global environment. And I really want the DX to be as one-shot as possible. The most brittle part of this seems to me to be the hardcoded assumptions about the paths, which I'm not entirely sure how to mitigate--see note below about the canonical home of this script.

The other main chunk of complexity is around the code to automatically assume the role, which is required since the tests require the config file to be written with assumed-role creds of the test execution role. Offloading that work from the customer (even if it were simply copy/paste commands) is complexity I'm willing to pay for, especially since the primary customer for that use case is me. :)

> Does this script belong in aws-amplify/amplify-ci-support as its canonical home?

Yes, definitely. The catch is that this needs to be invoked in platform-specific builds. Maybe we could do something like git submodules, or some other way of bootstrapping the dependency into the various places it needs to live? I'm open to suggestions. For Phase 1 though, it seems like a reasonable compromise to master this in the ci-support repo and manually copy it over to the platform-specific repos.